tests/test_shells.py

```diff
@@ -306,3 +308,47 @@ def test_distance_grid(cosmo: Cosmology) -> None:
 
 def test_combine() -> None:
     """Add unit tests for :func:`glass.combine`."""
+
+
+def test_radial_window_immutable() -> None:
+    """Checks the :class:`RadialWindow` class is immutable."""
+    wa = np.array([0.0, 1.0, 0.0])
+    za = np.array([0.0, 1.0, 2.0])
+    zeff = 1.0
+
+    w = glass.RadialWindow(za, wa, zeff)
+
+    with pytest.raises(dataclasses.FrozenInstanceError) as excinfo:
+        w.za = za
+
+    assert str(excinfo.value) == "cannot assign to field 'za'"
+
+    with pytest.raises(dataclasses.FrozenInstanceError) as excinfo:
+        w.wa = wa
+
+    assert str(excinfo.value) == "cannot assign to field 'wa'"
+
+    with pytest.raises(dataclasses.FrozenInstanceError) as excinfo:
```

Comment:
Does the `match` kwarg in `pytest.raises` not work for checking the error message here?

Reply:
I was following the docs https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Ah, their documentation is indeed confusing, but there is a subsection within that section - https://docs.pytest.org/en/stable/how-to/assert.html#matching-exception-messages

I don't have a strong preference, but to be consistent in our tests, this -
```py
with pytest.raises(dataclasses.FrozenInstanceError) as excinfo:
  w.wa = wa

assert str(excinfo.value) == "cannot assign to field 'wa'"
```
should change into -
```py
with pytest.raises(dataclasses.FrozenInstanceError, match="cannot assign to field 'wa'"):
  w.wa = wa
```

Reply:
Happy to do this. I vaguely remembered this was a thing, but the docs didn't have it first, so went with the `assert`.